pinot-common/src/main/java/com/linkedin/pinot/common/utils/retry/RetryPolicy.java

```diff
@@ -22,12 +22,13 @@
  * Retry policy, encapsulating the logic needed to retry an operation until it succeeds.
  */
 public interface RetryPolicy {
+
   /**
-   * Attempts to do the operation until it succeeds, aborting if an exception is thrown by the operation.
+   * Attempts to do the operation until it succeeds, aborting if an exception is thrown by the operation or number of
+   * attempts exhausted.
    *
    * @param operation The operation to attempt, which returns true on success and false on failure.
-   * @return true if the operation succeeded or false if the operation did not succeed within the retries specified by
-   * this retry policy
+   * @throws Exception
    */
-  boolean attempt(Callable<Boolean> operation);
+  void attempt(Callable<Boolean> operation) throws Exception;
```

Comment:
This should throw a more specific type of exception than `Exception` (eg. `AttemptsExceededException`, etc.).

Reply:
This could also throw arbitrary exceptions from the Callable